MDANSE/Src/MDANSE/Mathematics/Arithmetic.py

```diff
@@ -13,29 +13,50 @@
 #    You should have received a copy of the GNU General Public License
 #    along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
+from typing import List, Dict, Tuple
 import itertools
 
 import numpy as np
 
 
-def get_weights(props, contents, dim):
+def get_weights(props: Dict[str, float], contents: Dict[str, int], dim: int):
+    """Calculates the scaling factors to be applied to output datasets
+    of an analysis. Returns an dictionary of scaling factors, where the
+    chemical elements identifying each dataset are the keys.
+
+    Parameters
+    ----------
+    props : Dict[str, float]
+        Dictionary of values of an atom property for a selected object, averaged over atoms in that object
+    contents : Dict[str, int]
+        Dictionary of numbers of atoms in an object
+    dim : int
+        number of atom types in the label of the output datasets (e.g. 1 for "O", 2 for "CuCu")
+
+    Returns
+    -------
+    Tuple(Dict[Tuple[str], float], float)
+        Dictionary of scaling factors per dataset key, and a sum of all the factors
+    """
     normFactor = None
 
     weights = {}
 
     cartesianProduct = set(itertools.product(list(props.keys()), repeat=dim))
```

Comment:
Doesn't need to be cast as a `set`, `product` already unique.
```suggestion
    cartesianProduct = itertools.product(props, repeat=dim)
```